llm/cpp/llm.cpp

```diff
@@ -34,36 +34,14 @@ int main(int argc, char* argv[]) try {
     core.add_extension(USER_OV_EXTENSIONS_PATH);  // USER_OV_EXTENSIONS_PATH is defined in root CMakeLists.txt
     auto [input_ids, mask] = tokenize(core.compile_model(argv[2], "CPU").create_infer_request(), argv[4]);
     ov::InferRequest detokenizer = core.compile_model(argv[3], "CPU").create_infer_request();
-    std::shared_ptr<ov::Model> model = core.read_model(argv[1]);
-    std::map<size_t, ov::PartialShape> shapes = {
-        {0, ov::PartialShape{
-            -1, -1
-        }},
-        {1, ov::PartialShape{
-            -1, -1
-        }},
-        {2, ov::PartialShape{
-            -1, -1
-        }}
-    };
-    std::vector<ov::Output<ov::Node>> inputs = model->inputs();
-    for (size_t idx = 3; idx < inputs.size(); ++idx) {
-        ov::PartialShape shape = inputs.at(idx).get_partial_shape();
-        shape[0] = -1;
-        shapes.emplace(idx, shape);
-    }
-    model->reshape(shapes);
-    ov::InferRequest ireq = core.compile_model(model, "CPU", ov::cache_dir("llm-cache")).create_infer_request();
+    ov::InferRequest ireq = core.compile_model(argv[1], "CPU").create_infer_request();
     ireq.set_tensor("input_ids", input_ids);
     ireq.set_tensor("attention_mask", mask);
     ov::Tensor position_ids = ireq.get_tensor("position_ids");
     position_ids.set_shape(input_ids.get_shape());
     std::iota(position_ids.data<int64_t>(), position_ids.data<int64_t>() + position_ids.get_size(), 0);
-    for (size_t idx = 3; idx < inputs.size(); ++idx) {
-        ov::Shape shape = inputs.at(idx).get_partial_shape().get_min_shape();
-        shape.at(0) = 1;
-        ireq.get_input_tensor(idx).set_shape(shape);
-    }
+    ireq.get_tensor("beam_idx").set_shape({1});
```

Comment:
You can get a tensor once and use it in all places below, it would be shorter.

Reply:
It's no longer applicable for `beam_idx`, but I did it for `position_ids`